modules/cert-manager-prepare-cloud-credentials-aws-sts.adoc

```diff
@@ -0,0 +1,114 @@
+// Module included in the following assemblies:
+//
+// * security/cert_manager_operator/cert-manager-authenticate-aws.adoc
+
+:_content-type: PROCEDURE
+[id="cert-manager-prepare-cloud-credentials-aws-sts"]
+= Preparing cloud credentials for the {cert-manager-operator}
+
+You can prepare cloud credentials for the {cert-manager-operator} on the AWS Security Token Service (STS) cluster by using the `ccoctl` binary.
+
+.Prerequisites
+
+* You must extract and prepare the `ccoctl` binary.
+
+.Procedure
+
+. Create the `cert-manager` namespace by running the following command:
++
+[source,terminal]
+----
+$ oc create namespace cert-manager
+----
+
+. create a directory to store the `CredentialsRequest` custom resource (CR) of the {cert-manager-operator} by running the following command:
++
+[source,terminal]
+----
+$ mkdir credentials-request
+----
+
+. Change the directory by running the following command:
++
+[source,terminal]
+----
+$ cd credentials-request
+----
+
+. Create a `CredentialsRequest` resource YAML file, for example, `sample-credential-request.yaml`, as follows:
++
+[source,yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: cert-manager
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - "route53:GetChange"
+      effect: Allow
+      resource: "arn:aws:route53:::change/*"
+    - action:
+      - "route53:ChangeResourceRecordSets",
+      - "route53:ListResourceRecordSets"
+      effect: Allow
+      resource: "arn:aws:route53:::hostedzone/*"
+    - action:
+      - "route53:ListHostedZonesByName"
+      effect: Allow
+      resource: "*"
+  secretRef:
+    name: aws-creds
+    namespace: cert-manager
+  serviceAccountNames:
+  - cert-manager
+----
+
+. Create a `CredentialsRequest` resource by running the following command:
++
+[source,terminal]
+----
+$ oc create -f sample-credential-request.yaml
+----
+
+. Use the `ccoctl` tool to process `CredentialsRequest` objects of the {cert-manager-operator}, by running the following command:
++
+[source,terminal]
+----
+$ ccoctl aws create-iam-roles \
+    --name <name> --region=<aws_region> \
+    --credentials-requests-dir=<path-to-credrequests-dir> \
+    --identity-provider-arn <oidc-arn>
+----
+
+. Update the subscription object for {cert-manager-operator} by running the following command:
++
+[source,terminal]
+----
+$ oc -n cert-manager-operator patch subscription cert-manager-operator --type='merge' -p '{"spec":{"config":{"env":[{"name":"CLOUD_CREDENTIALS_SECRET_NAME","value":"aws-creds"}]}}}'
```

Comment:
We can skip this subscription patch step in AWS STS. The annotation that we did above should take care of everything.
Except we'd need to shuffle the sequence a bit, because the output of the ccoctl command will provide an ARN id string, that has to be used in the first annotation of the service account (`eks.amazonaws.com/role-arn`).
Again, feel free to refer to what is in https://github.com/openshift/cert-manager-operator/blob/master/docs/cloud_credentials.md#aws.

Reply:
I referred to the above link https://github.com/openshift/cert-manager-operator/blob/master/docs/cloud_credentials.md#aws and modified the "Preparing cloud credentials for the cert-manager Operator for Red Hat OpenShift" section. But I am wondering if the steps mentioned in the "Configuring the cert-manager Operator for Red Hat OpenShift to use prepared cloud credentials" section are complete! Can you please review if we need any other changes since I shuffled some commands?